var/spack/repos/builtin/packages/linux-external-modules/package.py

```diff
@@ -0,0 +1,339 @@
+# Copyright 2013-2024 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from shutil import copy2
+
+from spack.package import *
+
+
+class LinuxExternalModules(MakefilePackage):
+    """The Linux kernel provides services to start and stop programs,
+    handles the file system and other common "low-level" tasks that most
+    programs share, and schedules access to avoid conflicts when programs
+    try to access the same resource or device simultaneously. The kernel
+    has a modular design such that modules can be integrated as software
+    components. In this package, Linux has been configured to build out-of-tree
+    kernel modules."""
+
+    homepage = "https://github.com/torvalds/linux"
+    url = "https://cdn.kernel.org/pub/linux/kernel/v6.x/linux-6.10.2.tar.xz"
+    # url on how to build out-of-tree kernel modules
```

Comment:
Thanks for adding this as part of https://github.com/spack/spack/pull/45797/files#r1722558344. Can you mention that in particular section 5 describes how to use the modules?

Reply:
Added! It's section 2.1 that's the most important because you use the make syntax to make the external module, while pointing to the `linux-external-modules` install directory. I can still add section 5 - just lmk.

Reply:
I thought this package handled section 2 as part of `spack install linux-external-modules`.

The reason I thought it might be useful to refer to section 5, is that using these modules is (if I understand correctly) a bit different than how you would normally use a spack package: normally if you `spack install foo`, for example, you can `spack load foo` and then if there is a `foo` binary, it will get added to your PATH. 

To get your Linux OS to use the artifacts, I assume you'd have to set ``INSTALL_MOD_PATH=`spack location -i linux-external-modules` ``.